internals/setup.js

```diff
@@ -0,0 +1,398 @@
+      regex.test(stdout) ? resolve(false) : resolve(true)
+    })
+  })
+}
+
+/**
+ * Checks if this is a clone from our repo
+ * @returns {Promise<any>}
+ */
+function checkIfRepositoryIsAClone() {
+  return new Promise((resolve, reject) => {
+    exec('git remote -v', (err, stdout) => {
+      if (err) {
+        reject(new Error(err))
+      }
+
+      const isClonedRepo = stdout
+        .split(/\r?\n/)
+        .map(line => line.trim())
+        .filter(line => line.startsWith('origin'))
+        .filter(line => /kefranabg\/bento-starter\.git/.test(line)).length
+
+      resolve(!!isClonedRepo)
+    })
+  })
+}
+
+/**
+ * Remove the current Git repository
+ * @returns {Promise<any>}
+ */
+function removeGitRepository() {
+  return new Promise((resolve, reject) => {
+    try {
+      shell.rm('-rf', '.git/')
+      resolve()
+    } catch (err) {
+      reject(err)
+    }
+  })
+}
+
+/**
+ * Ask user if he wants to start with a new repository
+ * @returns {Promise<boolean>}
+ */
+function askUserIfWeShouldRemoveRepo() {
+  const NEW_REPOSITORY = 'NEW_REPOSITORY'
+  return inquirer
+    .prompt([
+      {
+        type: 'confirm',
+        message: 'Do you want to start with a new repository ?',
+        name: NEW_REPOSITORY
+      }
+    ])
+    .then(answers => !!answers[NEW_REPOSITORY])
+}
+
+/**
+ * Ask user for new origin for this repository. If provided, call git remote add origin
+ * @returns {Promise<any>}
+ */
+const askUserForNewRemote = () => {
+  const NEW_REMOTE = 'NEW_REMOTE'
+
+  return inquirer
+    .prompt([
+      {
+        type: 'input',
+        message:
+          'Enter new remote origin (ex: https://github.com/user/my-repo.git) for this repository [Enter to cancel]:  ',
+        name: NEW_REMOTE
+      }
+    ])
+    .then(answers => {
+      const origin = answers[NEW_REMOTE]
+      if (origin) {
+        const spinner = ora('Adding new remote to repository')
+        exec(`git remote add origin ${origin}`, error => {
+          if (error) {
+            spinner.fail('Add remote failed')
+            return false
+          } else {
+            spinner.success(
+              'New remote added, run `git push` to send initial commit to remote repository.'
+            )
+            return true
+          }
+        })
+      } else {
+        process.stdout.write(
+          'No remote added, run git remote add origin <url> to add one'
+        )
+        return false
+      }
+    })
+}
+
+/**
+ * Checks if we are under Git version control.
+ * If we are and this a clone of our repository the user is given a choice to
+ * either keep it or start with a new repository.
+ * @returns {Promise<boolean>}
+ */
+async function cleanCurrentRepository() {
+  const hasGitRepo = await hasGitRepository().catch(onError)
+
+  // We are not under Git version control. So, do nothing
+  if (hasGitRepo === false) {
+    return false
+  }
+
+  const isClone = await checkIfRepositoryIsAClone().catch(onError)
```

Comment:
I think it is not important that's a clone or not. We ask if the user want to start a new repository just after.

Reply:
@kefranabg  What do you think about that ?

Reply:
👍

Reply:
Hmm, actually I think it's a good thing to check if it's a clone. We to change remote only if it's a clone. When you fork you don't have to change the remote origin

Reply:
Well why not. But if it's a fork then it's probably in order to customize / add content / contribute to bento-starter, therefore you can start a new project based on that customized bento-starter fork. 

I could for instance fork bento-starter, customize it and clone it in order to create a new project. In my opinion we should at least look if the remote is from `/bento-starter\.git/` instead of `/kefranabg\/bento-starter\.git/`.